test/functional/core/fileio_spec.lua

```diff
@@ -139,3 +144,61 @@ describe('fileio', function()
   end)
 end)
 
+describe('skip do you really want to write on force write', function()
+  after_each(function()
+    os.remove('Xtest-overwrite-forced')
+  end)
+
+  it(':w! does not show "file has been changed" warning', function()
+    write_file("Xtest-overwrite-forced", 'foobar\n')
```

Comment:
Hmm, unfortunately test still fails on Windows. Then I have another suggestion
```suggestion
    write_file("Xtest-overwrite-forced", 'foobar')
    command('set nofixendofline')
```

Reply:
I do not have access to a windows machine to test this out but I think it might be related to the lack of a file extension. I'll commit that first and if the ci still fails after that I'll try your patch.